src/components/CreateArticle.jsx

```diff
@@ -19,112 +19,152 @@ class CreateArticle extends Component {
       photo,
       title,
       body,
-      image,
+      image
     };
+    this.track = 0;
     this.imageRef = React.createRef();
   }
 
-  async componentDidUpdate(prevProps, prevState) {
-    if ((this.props.publish && prevProps.publish === false)
-    || (this.props.draft && prevProps.draft === false)) {
-      let uploaded = true;
-      const { onSave, createArticle, publish } = this.props;
-      const {
-        photo,
-        image,
-        id,
-        title,
-        body
-      } = prevState;
-      if (title.trim().length < 3 || body.trim().length < 3) {
-        onSave();
+  async componentDidUpdate(prevProps) {
```

Comment:
Refactor this function to 👇 
```
async componentDidUpdate(prevProps) {
    const { title, body } = this.state;
    
    if (
      (prevProps.publish === false) &&
      (this.props.publish || this.props.draft)
    ) {
      if (title.trim().length < 3 || body.trim().length < 3) {
        this.props.onSave();
        this.props.addFlashMessage({
          type: 'error',
          text: 'title and body must not be less than 3 characters'
        });
      } else {
        await this.sendData();
      }
    } else {
      this.props.onSave();
    }
  }
```

What does `onSave` do? That method name is not very descriptive.

Reply:
using this `(prevProps.publish === false)` without `&& prevProps.publish === false` results to an infinit loop.